pwndbg/enhance.py

```diff
@@ -44,6 +44,14 @@ def format_small_int_pair(first: int, second: int) -> Tuple[str, str]:
 
 
 def int_str(value: int) -> str:
+    """
+    This function attempts to enhance the value of an int.
+    First, it checks if the integer is a stack canary.
+    Then, it attempts to interpret the bytes as a short strings
+    """
+    if value == pwndbg.commands.canary.canary_value()[0]:
```

Comment:
The canary_value() function is currently not cached. We should probably cache it, because this get called VERY often, and each time it will read from memory.

Reply:
It just needs be cached in a smart way - wait for the memory to be loaded in the first place, then assume it won't change during the program execution?

Reply:
The problem is: it may change ;D.

Reply:
Btw this calls a function from `pwndbg.commands.*` - those should likely almost never be called externally... We should probably move that canary function to some lib counterpart.

Reply:
This should also be gated somehow so that this is never called when:
1) debugging linux kernel
2) debugging embedded devices

etc.

Reply:
:0 didn't know the stack canary could change during the invocation of the program. Does the value pointed at by AT_RANDOM change? Or is there another way it changes?